homeassistant/components/yeelight/const.py

```diff
@@ -0,0 +1,40 @@
+"""Const for Yeelight."""
+
+DOMAIN = "yeelight"
+
+CONF_MODEL = "model"
+CONF_TRANSITION = "transition"
+CONF_SAVE_ON_CHANGE = "save_on_change"
+CONF_MODE_MUSIC = "use_music_mode"
+CONF_FLOW_PARAMS = "flow_params"
+CONF_CUSTOM_EFFECTS = "custom_effects"
+CONF_NIGHTLIGHT_SWITCH_TYPE = "nightlight_switch_type"
+CONF_NIGHTLIGHT_SWITCH = "nightlight_switch"
+
+DEFAULT_NAME = "Yeelight"
+DEFAULT_TRANSITION = 350
+DEFAULT_MODE_MUSIC = False
+DEFAULT_SAVE_ON_CHANGE = False
+DEFAULT_NIGHT_LIGHT_SWITCH = False
+
+CONF_MODELS = [
```

Comment:
I think its better to skip models list here, its additional work to synchronize supported model list with yeelight-lib. I've added method to do this https://gitlab.com/stavros/python-yeelight/-/blob/master/yeelight/main.py#L48 I think it would be better to get it from here, as model, to be effective ( declared temp range etc ) needs to be declared in yeelight-lib.

Reply:
Yes, I agree with that. As for the warning part, I'm still trying to figure out the best way of implement the config flow. Maybe I'll end up with a simple string field and add more detection and warning things in follow up PRs.